robot/impl/resource_manager.go

```diff
@@ -1019,14 +1026,52 @@ func (manager *resourceManager) markResourcesRemoved(
 func (manager *resourceManager) removeOrphanedResources(ctx context.Context,
 	rNames []resource.Name,
 ) {
-	// Ignore returned resources to close, as removeMarkedAndClose will handle.
 	manager.markResourcesRemoved(rNames, nil)
-	if _, err := manager.removeMarkedAndClose(ctx, nil); err != nil {
+	if err := manager.removeMarkedAndClose(ctx, nil); err != nil {
 		manager.logger.Errorw("error removing and closing marked resources",
 			"error", err)
 	}
 }
 
+// createConfig will create a config.Config based on the current state of the
+// resource graph, processManager and moduleManager.
+func (manager *resourceManager) createConfig() *config.Config {
+	conf := &config.Config{}
+
+	for _, resName := range manager.resources.Names() {
+		gNode, ok := manager.resources.Node(resName)
+		if !ok {
+			continue
+		}
+		resConf := gNode.Config()
+
+		// gocritic will complain that this if-else chain should be a switch, but
+		// it's really a mix of == and bool method checks.
+		//
+		//nolint: gocritic
+		if resName.API == client.RemoteAPI {
+			remoteConf, ok := resConf.ConvertedAttributes.(*config.Remote)
+			if !ok {
+				manager.logger.Errorw("ConvertedAttributes is not a config.Remote")
```

Comment:
w not needed but you could say what it is. Also you can use AssertType

Reply:
used `AssertType`; cool use of generics